sdk/core/azure_core_amqp/src/messaging.rs

```diff
@@ -586,77 +522,27 @@ impl From<AmqpMessageProperties> for AmqpList {
                     list.push(AmqpValue::ULong(message_id));
                 }
             }
-        } else if properties.user_id.is_some()
-            || properties.to.is_some()
-            || properties.subject.is_some()
-            || properties.reply_to.is_some()
-            || properties.correlation_id.is_some()
-            || properties.content_type.is_some()
-            || properties.content_encoding.is_some()
-            || properties.absolute_expiry_time.is_some()
-            || properties.creation_time.is_some()
-            || properties.group_id.is_some()
-            || properties.group_sequence.is_some()
-            || properties.reply_to_group_id.is_some()
-        {
+        } else {
```

Comment:
Tip: you could still simplify a lot of this conditional goo e.g.,

```rust
#[allow(dead_code)]
#[derive(Clone, Debug)]
enum AmqpValue {
    Null,
    String(String),
    UInt32(u32),
}

#[derive(Debug, Default)]
struct Options {
    pub s: Option<String>,
    pub u: Option<u32>,
}

fn main() {
    let options = Options {
        s: Some("string".to_string()),
        ..Default::default()
    };

    let mut values = vec![AmqpValue::Null; 2];
    values[0] = options
        .s
        .as_ref()
        .map_or_else(|| AmqpValue::Null, |v| AmqpValue::String(v.to_string()));
    values[1] = options
        .u
        .map_or_else(|| AmqpValue::Null, |v| AmqpValue::UInt32(v));

    println!("{options:?}");
}
```

Reply:
Oh, that is so nide.

One minor refinement:

```rust
values[1] = options.u.map_or(AmqpValue::Null, AmqpValue::Int32);
```

courtesy of GH copilot - testing this version right now, but reading the code, there's no reason it shouldn't work.

Reply:
Makes sense. Much like .NET delegates, if the signature matches you just pass in a ~~function~~ method.

Reply:
It's interesting. I think of `AmqpValue::UInt32(v)` as an initializer for `AmqpValue`, but apparently the actual semantics are that it's a function call to the `UInt32` function on `AmqpValue`.

Which is weird, but I guess it works.